unified-runtime/source/adapters/native_cpu/enqueue.cpp

```diff
@@ -66,8 +66,17 @@ static native_cpu::state getResizedState(const native_cpu::NDRDescT &ndr,
 UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
     ur_queue_handle_t hQueue, ur_kernel_handle_t hKernel, uint32_t workDim,
     const size_t *pGlobalWorkOffset, const size_t *pGlobalWorkSize,
-    const size_t *pLocalWorkSize, uint32_t numEventsInWaitList,
-    const ur_event_handle_t *phEventWaitList, ur_event_handle_t *phEvent) {
+    const size_t *pLocalWorkSize, uint32_t numPropsInLaunchPropList,
+    const ur_kernel_launch_property_t *launchPropList,
+    uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
+    ur_event_handle_t *phEvent) {
```

Comment:
It seems inconsistent with the general UR design that this dereferences a null pointer if `numPropsInLaunchPropList != 0` but `launchPropList == nullptr`, rather than returning an error code. Is that intentional?

Reply:
Potentially yes, we do have a weird situation where some validation needs to be in the adapters because a subset of sycl e2e tests rely on it being there whether you've got the layer enabled or not, but at least some of the checks below are likely superfluous.

Reply:
I've opened a PR to update the wording https://github.com/intel/llvm/pull/18724